zui/src/transforms/zui-to-typescript-next/index.test.ts

```diff
@@ -0,0 +1,594 @@
+        z.tuple([z.string(), z.number().describe('This is a number')]),
+      )
+
+    const typings = toTypescript(fn, { declaration: true })
+
+    expect(typings).toMatchWithoutFormatting(`
+      declare function fn(
+        arg0: {
+          a?: number;
+          /** This is B parameter */
+          b: string
+        },
+        /** This is a number */
+        arg1: number,
+        arg2: [string, /** This is a number */ number]
+      ): unknown;
+    `)
+    expect(typings).toBeValidTypeScript()
+  })
+
+  it('function with optional args', async () => {
+    const fn = z.function().title('fn').args(z.string().optional())
+    const typings = toTypescript(fn, { declaration: true })
+    expect(typings).toMatchWithoutFormatting('declare function fn(arg0?: string): unknown;')
+    expect(typings).toBeValidTypeScript()
+  })
+
+  it('string literals', async () => {
+    const typings = toTypescript(
+      z.union([z.literal('Hello, world!'), z.literal('Yoyoyoyo')]).describe('yoyoyo\nmultiline'),
+    )
+    expect(typings).toMatchWithoutFormatting(`
+      /**
+       * yoyoyo
+       * multiline
+       */
+      'Hello, world!' | 'Yoyoyoyo'
+    `)
+  })
+
+  it('function with named args', async () => {
+    const fn = z.function().title('fn').args(z.string().title('firstName').optional())
+    const typings = toTypescript(fn, { declaration: true })
+    expect(typings).toMatchWithoutFormatting('declare function fn(firstName?: string): unknown;')
+    expect(typings).toBeValidTypeScript()
+  })
+
+  it('mix of named and unnammed params', async () => {
+    const fn = z
+      .function()
+      .title('fn')
+      .args(z.string().title('firstName').optional(), z.number(), z.object({ a: z.string() }).title('obj'))
+    const typings = toTypescript(fn, { declaration: true })
+    expect(typings).toMatchWithoutFormatting(`
+      declare function fn(
+        firstName?: string,
+        arg1: number,
+        obj: { a: string }
+      ): unknown;
+    `)
+  })
+
+  it('nullables and optionals combined', async () => {
+    const fn = z
+      .function()
+      .title('fn')
+      .args(z.string().nullable().optional(), z.number().optional())
+      .returns(z.string().nullable().optional())
+
+    const typings = toTypescript(fn, { declaration: true })
+    expect(typings).toMatchWithoutFormatting(`
+      declare function fn(
+        arg0?: string | null,
+        arg1?: number
+      ): string | null | undefined;
+    `)
+  })
+})
+
+describe('objects', () => {
+  it('title mandatory to declare', async () => {
+    const obj = z.object({ a: z.number(), b: z.string() })
+    expect(() => toTypescript(obj, { declaration: true })).toThrowError()
+  })
+
+  it('normal object', async () => {
+    const obj = z.object({ a: z.number(), b: z.string() }).title('MyObject')
+
+    const typings = toTypescript(obj, { declaration: true })
+
+    expect(typings).toMatchWithoutFormatting('declare const MyObject: { a: number; b: string };')
```

Comment:
**nitpick:** I find it weird that the new typescript generator uses `declare` with variables instead of `type`. I was expecting:
```ts
export type MyFunction = (a: number) => string
export type MyObject = { a: number; b: string }
```

Reply:
it's simply easier to code as it works in more scenarios than export type !